src/30_Advanced/SeatMap.html

```diff
@@ -0,0 +1,175 @@
+    </noscript>
+
+  </head>
+  <body>
+    <!-- ## Seatmap -->
+    <!-- Here we are using an `amp-state` called `seatmap`.
+      `scale` variable is used to keep track of the scale of the content being zoomed using by `amp-pan-zoom`.
+      `zoom` variable is used to keep track of the seat map is in a zoomed state or not. This variable is used to
+    decide whether to show a + or - button.
+    `selectedSeats` variables keeps a  list of selected seat ids.  -->
+    <amp-state id="seatmap">
+      <script type="application/json">
+        {
+          "scale": 1,
+          "selectedSeats": [],
+          "zoom": 0
+        }
+      </script>
+    </amp-state>
+
+    <!-- You can implement a seat map by using `amp-list`, `amp-bind` and `amp-pan-zoom`.
+    `amp-list` is used to keep the availability of the seatmap up to date.
+    `amp-bind` is used to update the status of the seats: when a seat is selected or unselected we update the color and 
+    we add the seat id to the `seatmap.selectedSeats` variable. 
+    `amp-pan-zoom` is used to make possible to pan & zoom on mobile devices.
+    We are using the `transformEnd` action from [amp-pan-zoom](https://www.ampproject.org/docs/reference/components/amp-pan-zoom)
+    to update the value of transform.scale variable.
+    -->
+    <div>
+     <h1>Seat map</h1>
+     <div class="list-container">
+     <amp-list layout="fill" src="/json/seats.json" items="." single-item noloading>
+      <!--~
+      We display a placeholder for the seatmap by using server-side templating. This is possible because we have
+      the information about the layout of the theater.
+      ~-->
+      <div placeholder>
+        <svg class="opaque disabled seat-map-svg" viewBox="0 0 [[.Width]] [[.Height]]" preserveAspectRatio="xMidYMin slice">
+          [[range .Seats]] 
+           <rect class="seat unavailable" 
+                          x="[[.X]]" y="[[.Y]]" width="[[.Width]]" height="[[.Height]]"
+                          role="button"
+                          tabindex="0" rx="[[.Rx]]" ry="[[.Ry]]"/>   
+            [[end]]
+              
+        </svg>
+      </div>
+          <!--~
+            The seatmap with the availability of the seats is rendered via `amp-list` so that we make sure always the latest
+            data are available to the user.
+          ~-->
+          <template type="amp-mustache">
+            <amp-pan-zoom layout="fill" id="pan-zoom"
+              on="transformEnd:AMP.setState({seatmap: {scale: event.scale, x: event.x, y: event.y, zoom: event.scale <= 1 ? 0 : 1}})">
+              <svg class="seat-map-svg" preserveAspectRatio="xMidYMin slice" viewBox="0 0 {{width}} {{height}}">
+                {{#seats}}                
+                    <!--~
+                    If a seat is selected, when the user selects it again, it's removed from the seatmap.selectedSeats list
```

Comment:
* mark `seatmap.selectedSeats` as code
* What does this explanation apply to? 
* Rephrase: "Clicking on a seat will either select or deselect the seat depending on whether it's already been selected. This is implemented in the `on="tap:AMP.setState(...)"` action.